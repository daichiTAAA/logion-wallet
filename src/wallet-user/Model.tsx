
export interface TokenizationRequest {
    legalOfficerAddress: string,
    requesterAddress: string,
    requestedTokenName: string,
    bars: number,
}
