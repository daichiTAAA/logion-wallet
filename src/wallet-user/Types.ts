import { ColorTheme, rgbaToHex } from '../component/ColorTheme';

export const DARK_MODE: ColorTheme = {
    type: 'dark',
    shadowColor: rgbaToHex('#3b6cf4', 0.2),
    dashboard: {
        background: '#0c163d',
        foreground: '#ffffff',
    },
    sidebar: {
        background: '#152665',
        foreground: '#ffffff',
        activeItemBackground: rgbaToHex('#ffffff', 0.20),
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#ffffff', 0.6),
        foreground: '#ffffff',
        background: '#0c163d',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#152665',
        foreground: '#ffffff',
        link: '#ffffff',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondaryBackgroundColor: '#0c163d',
    },
    select: {
        background: rgbaToHex('#ffffff', 0.20),
        foreground: '#ffffff',
        menuBackgroundColor: '#152665',
        selectedOptionBackgroundColor: rgbaToHex('#ffffff', 0.20),
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: rgbaToHex('#ffffff', 0.20),
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    }
};