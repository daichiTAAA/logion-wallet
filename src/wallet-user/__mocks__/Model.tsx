import { TokenizationRequest } from "../../common/types/ModelTypes";
import { CreateTokenRequest } from "../Model";

export async function createTokenRequest(request: CreateTokenRequest): Promise<TokenizationRequest> {
    return Promise.resolve({
        id: "1234",
        legalOfficerAddress: request.legalOfficerAddress,
        requesterAddress: request.requesterAddress,
        requestedTokenName: request.requestedTokenName,
        bars: request.bars,
        status: "PENDING"
    });
}
