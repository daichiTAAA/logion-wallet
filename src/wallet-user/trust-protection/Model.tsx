import axios from "axios";
import {Moment} from "moment";
import {DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER, ProtectionRequest} from "../../legal-officer/Types";
import Identity from '../../component/types/Identity';
import PostalAddress from '../../component/types/PostalAddress';

export interface CreateProtectionRequest {
    requesterAddress: string,
    userIdentity: Identity,
    userPostalAddress: PostalAddress,
    legalOfficerAddresses: string[],
    signature: string,
    signedOn: Moment
}

export type LegalOfficerDecisionStatus = "PENDING" | "REJECTED" | "ACCEPTED";

export interface LegalOfficerDecision {
    legalOfficerAddress: string,
    status: LegalOfficerDecisionStatus
}

export async function createProtectionRequest(request: CreateProtectionRequest): Promise<ProtectionRequest> {
    const response = await axios.post("/api/protection-request", request);
    return response.data;
}

function fakeAddress(i: number) {
    return "1234567890-abcd-" + i;
}

export const legalOfficers = [
    {name: "Patrick", address: DEFAULT_LEGAL_OFFICER},
    {name: "Guillaume", address: ANOTHER_LEGAL_OFFICER},
    {name: "Alain", address: fakeAddress(5)},
    {name: "Ronald", address: fakeAddress(6)},
    {name: "Thibaut", address: fakeAddress(7)},
    {name: "Jona", address: fakeAddress(8)},
    {name: "Carlos", address: fakeAddress(9)},
    {name: "Ilias", address: fakeAddress(10)},
    {name: "Pedroso", address: fakeAddress(11)},
    {name: "Marc", address: fakeAddress(12)},
]

export function legalOfficerName(address: string): string {
    const legalOfficer = legalOfficers.find(legalOfficer => legalOfficer.address === address);
    return legalOfficer === undefined ? "Unknown" : legalOfficer.name;
}

