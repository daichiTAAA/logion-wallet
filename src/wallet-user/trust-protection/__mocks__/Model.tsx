export function checkActivation() {
    return Promise.resolve();
}
