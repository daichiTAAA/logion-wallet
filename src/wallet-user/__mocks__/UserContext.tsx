import {DEFAULT_LEGAL_OFFICER, TokenizationRequest} from "../../legal-officer/Types";
import {TEST_WALLET_USER} from "../Model.test";
import {ApiPromise} from '@polkadot/api';
import {ProtectionRequest} from "../../legal-officer/Types";
import { DARK_MODE } from '../Types';

export let createTokenRequest = () => null;

export let createdTokenRequest: TokenizationRequest | null = null;

export let pendingTokenizationRequests: TokenizationRequest[] | null = null;

export let acceptedTokenizationRequests: TokenizationRequest[] | null = null;

export let rejectedTokenizationRequests: TokenizationRequest[] | null = null;

export let refreshRequests = jest.fn();

export let createProtectionRequest = () => null;

export let pendingProtectionRequests: ProtectionRequest[] | null = null;

export let acceptedProtectionRequests: ProtectionRequest[] | null = null;

export let api = new ApiPromise();

export let recoveryConfig = null;

export let setUserAddress = jest.fn();

export let userAddress = TEST_WALLET_USER;

export function useUserContext() {
    return {
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        userAddress,
        createTokenRequest,
        createdTokenRequest,
        pendingTokenizationRequests,
        acceptedTokenizationRequests,
        rejectedTokenizationRequests,
        refreshRequests,
        createProtectionRequest,
        pendingProtectionRequests,
        acceptedProtectionRequests,
        recoveryConfig,
        api,
        setUserAddress,
        colorTheme: DARK_MODE,
    };
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setCreatedTokenRequest(request: TokenizationRequest) {
    createdTokenRequest = request;
}

export function setPendingRequests(requests: TokenizationRequest[]) {
    pendingTokenizationRequests = requests;
}

export function setRejectedRequests(requests: TokenizationRequest[]) {
    rejectedTokenizationRequests = requests;
}

export function setAcceptedRequests(requests: TokenizationRequest[]) {
    acceptedTokenizationRequests = requests;
}

export function setRefreshRequests(func: any) {
    refreshRequests = func;
}

export function setPendingProtectionRequests(requests: ProtectionRequest[] | null) {
    pendingProtectionRequests = requests;
}

export function setAcceptedProtectionRequests(requests: ProtectionRequest[] | null) {
    acceptedProtectionRequests = requests;
}

export function setCreateProtectionRequest(callback: any) {
    createProtectionRequest = callback;
}

export function setRecoveryConfig(config: any) {
    recoveryConfig = config;
}

export function UserContextProvider() {
    return (
        <div></div>
    );
}

export function setContextUserAddress(address: string) {
    userAddress = address;
}