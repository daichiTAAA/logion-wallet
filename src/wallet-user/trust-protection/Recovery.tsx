import React from 'react';

import { useUserContext } from "../UserContext";
import { isRecovery } from "./Model";

import GoToTrustProtection from './GoToTrustProtection';
import CreateProtectionRequestForm from "./CreateProtectionRequestForm";
import ProtectionRecoveryRequest from './ProtectionRecoveryRequest';
import RecoveryProcess from './RecoveryProcess';

export default function Recovery() {
    const { pendingProtectionRequests, acceptedProtectionRequests, recoveryConfig, recoveredAddress } = useUserContext();

    if (pendingProtectionRequests === null || acceptedProtectionRequests === null || recoveryConfig === null
            || recoveredAddress === undefined) {
        return null;
    }

    const requests = pendingProtectionRequests.concat(acceptedProtectionRequests);
    const goToTrustProtection = ((pendingProtectionRequests.length > 0) && !isRecovery(pendingProtectionRequests[0]))
        || ((acceptedProtectionRequests.length > 0) && !isRecovery(acceptedProtectionRequests[0]));

    if(goToTrustProtection) {
        console.log(pendingProtectionRequests);
        console.log(acceptedProtectionRequests);
        return <GoToTrustProtection />;
    } else if(pendingProtectionRequests.length > 0) {
        return <ProtectionRecoveryRequest requests={ requests } type='pending' />;
    } else if(acceptedProtectionRequests.length === 2) {
        if(recoveryConfig.isEmpty) {
            return <ProtectionRecoveryRequest requests={ requests } type='accepted' />;
        } else if(recoveryConfig.isSome && recoveredAddress === null) {
            return <ProtectionRecoveryRequest requests={ requests } type='activated' />;
        } else {
            return <RecoveryProcess />;
        }
    } else {
        return <CreateProtectionRequestForm isRecovery={ true } />;
    }
}
