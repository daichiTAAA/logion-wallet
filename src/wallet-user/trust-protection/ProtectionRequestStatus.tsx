import React, { useCallback, useState } from 'react';
import Button from 'react-bootstrap/Button';
import { Link } from 'react-router-dom';

import { useLogionChain, Unsubscriber, ISubmittableResult, replaceUnsubscriber } from '../../logion-chain';
import { createRecovery } from '../../logion-chain/Recovery';
import ExtrinsicSubmissionResult from '../../legal-officer/ExtrinsicSubmissionResult';

import {useUserContext} from "../UserContext";
import {legalOfficerName} from "./Model";
import {ProtectionRequest} from "../../legal-officer/Types";
import { ACCOUNT_PATH } from '../UserRouter';

import CreateProtectionRequestForm from "./CreateProtectionRequestForm";

export default function ProtectionRequestStatus() {
    const { api } = useLogionChain();
    const { userAddress, pendingProtectionRequests, acceptedProtectionRequests, recoveryConfig, refreshRequests } = useUserContext();
    const [ activationResult, setActivationResult ] = useState<ISubmittableResult | null>(null);
    const [ activationError, setActivationError ] = useState<any>(null);
    const [ activationUnsubscriber, setActivationUnsubscriber ] = useState<Unsubscriber | null>(null);

    const activateProtection = useCallback((request: ProtectionRequest) => {
        setActivationResult(null);
        setActivationError(null);
        (async function() {
            const unsubscriber = createRecovery({
                api: api!,
                signerId: userAddress,
                callback: setActivationResult,
                errorCallback: setActivationError,
                legalOfficers: request.decisions.map(decision => decision.legalOfficerAddress),
            });
            await replaceUnsubscriber(activationUnsubscriber, setActivationUnsubscriber, unsubscriber);
            refreshRequests!();
        })();
    }, [ api, userAddress, refreshRequests, activationUnsubscriber, setActivationUnsubscriber ]);

    if (pendingProtectionRequests === null || acceptedProtectionRequests === null || recoveryConfig === null) {
        return null;
    }

    if(pendingProtectionRequests.length > 0) {
        const createdProtectionRequest: ProtectionRequest = pendingProtectionRequests[0];

        return (
            <>
                <h1>Protection request status</h1>
                <p>
                    Your Logion Trust Protection request ({createdProtectionRequest.id}) has been submitted. A Legal Officer
                    will contact you as soon as possible to finalize the KYC process. Please note that, after the successful
                    completion of one of your Legal Officer approval processes, you will be able to use all features
                    provided by your logion account dashboard.
                </p>
                <ul>
                {
                    createdProtectionRequest.decisions.map(decision => (
                        <li key={decision.legalOfficerAddress}>
                            {legalOfficerName(decision.legalOfficerAddress)} ({decision.legalOfficerAddress}):
                            &bnsp;{decision.status}
                        </li>
                    ))
                }
                </ul>
                <Link to={ ACCOUNT_PATH }>Back to account</Link>
            </>
        );
    } else if(acceptedProtectionRequests.length > 0) {
        const createdProtectionRequest: ProtectionRequest = acceptedProtectionRequests[0];

        if(recoveryConfig!.isEmpty) {
            return (
                <>
                    <h1>Protection request status</h1>
                    <p>
                        Your Logion Trust Protection request ({createdProtectionRequest.id}) has been accepted by your
                        Legal Officers. You can now activate your protection.
                    </p>
                    {
                        activationResult === null &&
                        <p>
                            <Button onClick={() => activateProtection(createdProtectionRequest)}>Activate</Button>
                        </p>
                    }
                    {
                        (activationResult !== null || activationError !== null) &&
                        <ExtrinsicSubmissionResult
                            result={activationResult}
                            error={activationError}
                            successMessage="Protection successfully activated."
                        />
                    }
                    <Link to={ ACCOUNT_PATH }>Back to account</Link>
                </>
            );
        } else {
            return (
                <>
                    <h1>Protection request status</h1>
                    <p>
                        Your Logion Trust Protection is active. You are now protected by
                    </p>
                    <ul>
                    {
                        createdProtectionRequest.decisions.map(decision => (
                            <li key={decision.legalOfficerAddress}>
                                {legalOfficerName(decision.legalOfficerAddress)} ({decision.legalOfficerAddress})
                            </li>
                        ))
                    }
                    </ul>
                    <Link to={ ACCOUNT_PATH }>Back to account</Link>
                </>
            );
        }
    } else {
        return <CreateProtectionRequestForm />;
    }
}
