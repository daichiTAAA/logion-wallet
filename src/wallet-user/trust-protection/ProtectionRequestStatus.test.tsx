jest.mock('../UserContext');
jest.mock('../../logion-chain');
jest.mock('../../logion-chain/Recovery');

import moment from 'moment';
import { render, waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { createRecovery } from '../../logion-chain/Recovery';

import { setPendingProtectionRequests, setAcceptedProtectionRequests, setRecoveryConfig } from "../UserContext";
import { shallowRender } from "../../tests";
import React from "react";
import ProtectionRequestStatus from "./ProtectionRequestStatus";
import { ProtectionRequest } from '../../legal-officer/Types';
import { DEFAULT_IDENTITY } from '../../component/Identity.test';
import { DEFAULT_ADDRESS } from '../../component/PostalAddress.test';
import {TEST_WALLET_USER} from "../Model.test";

test("renders empty", () => {
    const tree = shallowRender(<ProtectionRequestStatus/>)
    expect(tree).toMatchSnapshot();
});

test("renders pending protection request", () => {
    const requests: ProtectionRequest[] = [
        {
            id: '2eb5f71c-7f31-44b5-9390-c3bf56501880',
            requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
            decisions: [
                {
                    legalOfficerAddress: '5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty',
                    status: 'PENDING',
                    rejectReason: null,
                    decisionOn: null
                },
                {
                    legalOfficerAddress: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY',
                    status: 'PENDING',
                    rejectReason: null,
                    decisionOn: null
                }
            ],
            userIdentity: DEFAULT_IDENTITY,
            userPostalAddress: DEFAULT_ADDRESS,
            createdOn: moment('2021-06-10T13:48:00.000Z').toISOString(),
        }
    ];
    setPendingProtectionRequests(requests);
    const tree = shallowRender(<ProtectionRequestStatus/>)
    expect(tree).toMatchSnapshot();
});

test("renders accepted protection request", () => {
    setPendingProtectionRequests([]);
    const requests: ProtectionRequest[] = [
        {
            id: '2eb5f71c-7f31-44b5-9390-c3bf56501880',
            requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
            decisions: [
                {
                    legalOfficerAddress: '5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty',
                    status: 'ACCEPTED',
                    rejectReason: null,
                    decisionOn: null
                },
                {
                    legalOfficerAddress: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY',
                    status: 'ACCEPTED',
                    rejectReason: null,
                    decisionOn: null
                }
            ],
            userIdentity: DEFAULT_IDENTITY,
            userPostalAddress: DEFAULT_ADDRESS,
            createdOn: moment('2021-06-10T13:48:00.000Z').toISOString(),
        }
    ];
    setAcceptedProtectionRequests(requests);
    setRecoveryConfig({
        isEmpty: true
    })
    const tree = shallowRender(<ProtectionRequestStatus/>)
    expect(tree).toMatchSnapshot();
});

test("Activation of accepted protection request", async () => {
    setPendingProtectionRequests([]);
    const requester = TEST_WALLET_USER;
    const legalOfficers = ['5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty', '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY'];
    const requests: ProtectionRequest[] = [
        {
            id: '2eb5f71c-7f31-44b5-9390-c3bf56501880',
            requesterAddress: requester,
            decisions: [
                {
                    legalOfficerAddress: legalOfficers[0],
                    status: 'ACCEPTED',
                    rejectReason: null,
                    decisionOn: null
                },
                {
                    legalOfficerAddress: legalOfficers[1],
                    status: 'ACCEPTED',
                    rejectReason: null,
                    decisionOn: null
                }
            ],
            userIdentity: DEFAULT_IDENTITY,
            userPostalAddress: DEFAULT_ADDRESS,
            createdOn: moment('2021-06-10T13:48:00.000Z').toISOString(),
        }
    ];
    setAcceptedProtectionRequests(requests);
    setRecoveryConfig({
        isEmpty: true
    })

    render(<ProtectionRequestStatus/>);
    const activateButton = screen.getByRole('button');
    userEvent.click(activateButton);

    await waitFor(() => expect(createRecovery).toBeCalledWith(expect.objectContaining({
        api: expect.anything(),
        signerId: requester,
        callback: expect.anything(),
        errorCallback: expect.anything(),
        legalOfficers,
    })));
});

test("renders protected", () => {
    setPendingProtectionRequests([]);
    const requests: ProtectionRequest[] = [
        {
            id: '2eb5f71c-7f31-44b5-9390-c3bf56501880',
            requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
            decisions: [
                {
                    legalOfficerAddress: '5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty',
                    status: 'ACCEPTED',
                    rejectReason: null,
                    decisionOn: null
                },
                {
                    legalOfficerAddress: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY',
                    status: 'ACCEPTED',
                    rejectReason: null,
                    decisionOn: null
                }
            ],
            userIdentity: DEFAULT_IDENTITY,
            userPostalAddress: DEFAULT_ADDRESS,
            createdOn: moment('2021-06-10T13:48:00.000Z').toISOString(),
        }
    ];
    setAcceptedProtectionRequests(requests);
    setRecoveryConfig({
        isEmpty: false
    })
    const tree = shallowRender(<ProtectionRequestStatus/>)
    expect(tree).toMatchSnapshot();
});
