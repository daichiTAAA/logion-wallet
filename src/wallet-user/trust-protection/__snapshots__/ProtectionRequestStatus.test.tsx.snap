// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context initially fetches requests 1`] = `
Array [
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Line 1
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Place de le République Française, 10"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Line 2
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="boite 15"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Postal code
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="4000"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      City
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Liège"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Country
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Belgium"
    />
  </div>,
]
`;

exports[`renders 1`] = `
Array [
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      First name
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="John"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Last name
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="Doe"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      E-mail
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="john.doe@logion.network"
    />
  </div>,
  <div
    className="form-group"
  >
    <label
      className="form-label"
    >
      Phone number
    </label>
    <input
      className="form-control"
      readOnly={true}
      type="text"
      value="+1234"
    />
  </div>,
]
`;

exports[`renders accepted protection request 1`] = `
<React.Fragment>
  <h2>
    My Legal Officers
  </h2>
  <p>
    Your Logion Trust Protection request (
    2eb5f71c-7f31-44b5-9390-c3bf56501880
    ) has been accepted by your Legal Officers. You can now activate your protection.
  </p>
  <ul>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
            "details": "SELARL ADRASTEE
Gare des Brotteaux
14, place Jules Ferry
69006 LYON",
            "name": "Guillaume",
          }
        }
      />
    </li>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
            "details": "Modero Bruxelles
Central administration 
Nijverheidslaan 1
B - 1853 Grimbergen (Strombeek-Bever)",
            "name": "Patrick",
          }
        }
      />
    </li>
  </ul>
  <p>
    <Button
      active={false}
      data-testid="btnActivate"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Activate
    </Button>
  </p>
</React.Fragment>
`;

exports[`renders empty 1`] = `null`;

exports[`renders pending protection request 1`] = `
<React.Fragment>
  <h2>
    Protection request status
  </h2>
  <p>
    Your Logion Trust Protection request (
    2eb5f71c-7f31-44b5-9390-c3bf56501880
    ) has been submitted. A Legal Officer will contact you as soon as possible to finalize the KYC process. Please note that, after the successful completion of one of your Legal Officer approval processes, you will be able to use all features provided by your logion account dashboard.
  </p>
  <ul>
    <li>
      Guillaume
       (
      5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty
      ): 
      PENDING
    </li>
    <li>
      Patrick
       (
      5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
      ): 
      PENDING
    </li>
  </ul>
</React.Fragment>
`;

exports[`renders protected 1`] = `
<React.Fragment>
  <h2>
    My Legal Officers
  </h2>
  <p>
    Your Logion Trust Protection is active. You are now protected by
  </p>
  <ul>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty",
            "details": "SELARL ADRASTEE
Gare des Brotteaux
14, place Jules Ferry
69006 LYON",
            "name": "Guillaume",
          }
        }
      />
    </li>
    <li>
      <LegalOfficerInfo
        legalOfficer={
          Object {
            "address": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
            "details": "Modero Bruxelles
Central administration 
Nijverheidslaan 1
B - 1853 Grimbergen (Strombeek-Bever)",
            "name": "Patrick",
          }
        }
      />
    </li>
  </ul>
</React.Fragment>
`;
